### PR TITLE
Add local file support, separate and parametrize queries

### PR DESCRIPTION
**Pull request by mvs5465**
Allows users to run the function against local files (instead of S3) by setting these two properties:

```
USE_LOCAL_DATA = True # whether to load data from S3 (false) or locally (true)
LOCAL_DATA_REPOSITORY = "s3data/usdot-its-cvpilot-public-data" # path to local directory containing s3 data
```

Also moved queries into separate class so they can be passed as parameters (passing a function pointer).

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -75,23 +78,38 @@ def perform_query_1(s3_client, s3_file_list):
...
-    if response.get('Contents'):
-        [filenames.append(item['Key']) for item in response.get('Contents')]
-    while response.get('NextContinuationToken'):
-        response = list_s3_objects(s3_client, prefix_string, response.get('NextContinuationToken'))
+    if USE_LOCAL_DATA:
+        try:
+            files_and_directories = glob.glob(LOCAL_DATA_REPOSITORY+"/"+prefix_string+"/**/*", recursive=True)
+            files_only = []
+            for filepath in files_and_directories:
+                if os.path.isfile(filepath):
```
Comment (hmusavi) [inline main.py:122]: This seems to return file only in the current directory. Don't we need an `else` part to recursively scan the directories?

**Review by hmusavi — APPROVED** [review]
Looks good with one comment/question.

File: main.py (hunk 1 above)
Comment (mvs5465) [reply main.py:122]: `glob.glob()` recursively returns a list of everything that satisfies the constraints given, which in this case returns both directories and files. This if statement check makes sure that only files are returned, not directories.

Comment (hmusavi) [reply main.py:122]: So does it return files inside those directories too? If so, then the logic is correct. If not, we will be skipping those files without a recursive `else` part.

Comment (mvs5465) [reply main.py:122]: Yes the glob.glob recursively returns all files and folders inside all subdirectories as well. It returns the same file count as actual S3 query, so I believe the logic to be correct.

**Pull request merged**